internal/zinc-apiinfo/src/main/scala/sbt/internal/inc/ClassToAPI.scala

```diff
@@ -548,56 +548,18 @@ object ClassToAPI {
   }.toMap
   def primitive(name: String): api.Type = PrimitiveRefs(name)
 
-  // Workarounds for https://github.com/sbt/sbt/issues/1035
-  //   these catch the GenericSignatureFormatError and return the erased type
-
-  private[this] def returnType(f: Field): Type =
-    try f.getGenericType
-    catch {
-      case _: GenericSignatureFormatError => f.getType
-    }
-  private[this] def parameterTypes(c: Constructor[_]): Array[Type] =
-    try c.getGenericParameterTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(c.getParameterTypes)
-    }
-  private[this] def exceptionTypes(c: Constructor[_]): Array[Type] =
-    try c.getGenericExceptionTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(c.getExceptionTypes)
-    }
-  private[this] def parameterTypes(m: Method): Array[Type] =
-    try m.getGenericParameterTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(m.getParameterTypes)
-    }
-  private[this] def returnType(m: Method): Type =
-    try m.getGenericReturnType
-    catch {
-      case _: GenericSignatureFormatError => m.getReturnType
-    }
-  private[this] def exceptionTypes(m: Method): Array[Type] =
-    try m.getGenericExceptionTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(m.getExceptionTypes)
-    }
-
-  private[this] def typeParameterTypes[T](m: Constructor[T]): Array[TypeVariable[Constructor[T]]] =
-    try m.getTypeParameters
-    catch {
-      case _: GenericSignatureFormatError => new Array(0)
-    }
-  private[this] def typeParameterTypes[T](m: Class[T]): Array[TypeVariable[Class[T]]] =
-    try m.getTypeParameters
-    catch {
-      case _: GenericSignatureFormatError => new Array(0)
-    }
-  private[this] def typeParameterTypes(m: Method): Array[TypeVariable[Method]] =
-    try m.getTypeParameters
-    catch {
-      case _: GenericSignatureFormatError => new Array(0)
-    }
-
-  private[this] def convert(classes: Array[Class[_]]): Array[Type] =
-    classes.asInstanceOf[Array[Type]] // ok: treat Arrays as read-only
+  @inline private[this] def returnType(f: Field): Type = f.getGenericType
```

Comment:
That's a benchmark of numerical computations. Those kind of benchmarks are really misleading and easy to optimize efficiently by the Hotspot, probably with not even inlining but also loop unfolding et al. To add to that, it doesn't say anything about the benchmark setup and 2.12.3 improved the optimizer and inliner, making those benchmarks (for 2.12.0-M2) useless. So I think that link has no relevance to the scenario we're dealing with here.
  
By the by, this scenario is clearly beneficial for inlining because we only have one call-site, and it's in the same class definition (unlike the benchmark you mention). The benefit is that we don't pay the cost of invoking these functions, while still making them logically independent if in the future we want to add more logic to them, like a fix for https://github.com/sbt/zinc/issues/389.
  
I think we would agree that Zinc is more similar to Scalac than to that benchmark. It is reasonable to think `@inline` here is beneficial, given how beneficial it is for the Scala compiler. If you want to keep having this discussion, I suggest you have it with the Scala team. They will give you better insights than I will.
  
That being said, it would be good that you comment on the core issue that this patch is addressing.